File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>


using namespace std;

int solution(int n, int l, int p) {
    int ans = 0;
    for(int i=1;i<=n;i++){
        int cnt=0;
        for(int j=1;j*j<=i;j++){  
            if(i%j==0) {
                cnt+=1;
                if(j*j<i) cnt+=1;
            }
        }
        ans += (cnt > l) ? p : cnt; 
    }
    return ans;
}

File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.java
import java.util.stream.IntStream;

class Solution {
  public int solution(int number, int limit, int power) {
        return IntStream.rangeClosed(1, number)
                        .map(i -> {
                            int divisorCnt = getDivisorCnt(i);
                            return (divisorCnt > limit) ? power : divisorCnt;
                        }).sum();
    }

    private int getDivisorCnt(int num){
        int cnt = 0;
        for(int i=1;i*i<=num;i++){
            if(i*i == num) cnt+=1;
            else if(num%i==0) cnt+=2;
        }
        return cnt;
    }
}